File: frontend/src/order/object/core/OrderChatMessage.ts
import type {
  ChatMessageId,
  DescriptionText,
  IsoDateTime,
  PersonName,
  Role,
} from '@/shared/object/domain/DomainObjects'

export type OrderChatMessage = {
  id: ChatMessageId
  senderRole: Role
  senderName: PersonName
  body: DescriptionText
  sentAt: IsoDateTime
}

File: backend/src/main/scala/services/order/objects/core/OrderChatMessage.scala
package domain.order

import domain.shared.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import domain.shared.*

final case class OrderChatMessage(
    id: ChatMessageId,
    senderRole: UserRole,
    senderName: PersonName,
    body: DisplayText,
    sentAt: IsoDateTime,
)
object OrderChatMessage:
  given Encoder[OrderChatMessage] = deriveEncoder
  given Decoder[OrderChatMessage] = deriveDecoder
